component/vshn_postgres.jsonnet

```diff
@@ -638,6 +638,212 @@ local clusterRestoreConfig = {
+      },
+    },
+  },
+  patches: [
+    comp.FromCompositeFieldPathWithTransformSuffix('metadata.labels[crossplane.io/composite]', 'metadata.name', 'maintenancerolebinding'),
+    comp.FromCompositeFieldPathWithTransformSuffix('metadata.labels[crossplane.io/composite]', 'spec.forProvider.manifest.metadata.name', 'maintenancerolebinding'),
+    comp.FromCompositeFieldPathWithTransformPrefix('metadata.labels[crossplane.io/composite]', 'spec.forProvider.manifest.metadata.namespace', 'vshn-postgresql'),
+  ],
+};
+
+local convertToCron() = [
+  // This function produces patches, that will convert dayOdWeek and timeOfDay
+  // to a proper cron string. It does that by using maps and regex. As well as
+  // environment patches.
+  {
+    type: 'FromCompositeFieldPath',
+    fromFieldPath: 'spec.parameters.maintenance.dayOfWeek',
+    toFieldPath: 'metadata.annotations[dayOfWeek]',
+    transforms: [
+      {
+        type: 'map',
+        map: {
+          monday: '1',
+          tuesday: '2',
+          wednesday: '3',
+          thursday: '4',
+          friday: '5',
+          saturday: '6',
+          sunday: '0',
+        },
+      },
+    ],
+  },
+  {
+    type: 'FromCompositeFieldPath',
+    fromFieldPath: 'spec.parameters.maintenance.timeOfDay',
+    toFieldPath: 'metadata.annotations[hour]',
+    transforms: [
+      {
+        type: 'string',
+        string: {
+          type: 'Regexp',
+          regexp: {
+            match: '(\\d+):(\\d+):.*',
+            group: 1,
+          },
+        },
+      },
+    ],
+  },
+  {
+    type: 'FromCompositeFieldPath',
+    fromFieldPath: 'spec.parameters.maintenance.timeOfDay',
+    toFieldPath: 'metadata.annotations[minute]',
+    transforms: [
+      {
+        type: 'string',
+        string: {
+          type: 'Regexp',
+          regexp: {
+            match: '(\\d+):(\\d+):.*',
+            group: 2,
+          },
+        },
+      },
+    ],
+  },
+  {
+    type: 'ToEnvironmentFieldPath',
+    fromFieldPath: 'metadata.annotations[minute]',
+    toFieldPath: 'maintenance.minute',
+  },
+  {
+    type: 'ToEnvironmentFieldPath',
+    fromFieldPath: 'metadata.annotations[hour]',
+    toFieldPath: 'maintenance.hour',
+  },
+  {
+    type: 'ToEnvironmentFieldPath',
+    fromFieldPath: 'metadata.annotations[dayOfWeek]',
+    toFieldPath: 'maintenance.dayOfWeek',
+  },
+  {
+    type: 'CombineFromEnvironment',
+    toFieldPath: 'spec.forProvider.manifest.spec.schedule',
+    combine: {
+      variables: [
+        { fromFieldPath: 'maintenance.minute' },
+        { fromFieldPath: 'maintenance.hour' },
+        { fromFieldPath: 'maintenance.dayOfWeek' },
+      ],
+      strategy: 'string',
+      string: {
+        fmt: '%s %s * * %s',
+      },
+    },
+  },
+];
+
+local maintenanceJob = {
+  base: comp.KubeObject('batch/v1', 'CronJob') + {
+    spec+: {
+      forProvider+: {
+        manifest+: {
+          metadata+: {
+            namespace: pgParams.controlNamespace,
+          },
```

Comment:
Nitpick: This is a bit misleading as we patch it anyways and it actually shouldn't run in the control namespace

Reply:
Whoops, you're right, that's probably a copy paste error. Good catch.